frame/identity/src/lib.rs

```diff
@@ -473,40 +479,65 @@ decl_module! {
 		/// - `O(R)` where `R` registrar-count (governance-bounded).
 		/// - One storage mutation (codec `O(R)`).
 		/// - One event.
+		/// - Benchmark: 78.71 + R * 0.965 µs (min square analysis)
 		/// # </weight>
-		#[weight = SimpleDispatchInfo::FixedNormal(MINIMUM_WEIGHT)]
-		fn add_registrar(origin, account: T::AccountId) {
+		#[weight = FunctionOf(
+			|(_, &old_count): (&T::AccountId, &u32)| {
+				T::DbWeight::get().reads_writes(1, 1)
+				+ 78_710_000 // constant
+				+ 965_000 * (old_count as Weight) // R
+			},
+			DispatchClass::Normal,
+			true
+		)]
+		fn add_registrar(origin, account: T::AccountId, old_registrar_count: u32) -> DispatchResult {
```

Comment:
I think in this case, we can initially charge for `MaxRegistrarCount` and then refund the difference since we will know what the registrar len is by the end of this extrinsic, thus the new param is not needed.